### ISSUE TITLE
Non-deterministic predictions with `sample_new_groups=True` despite setting `random_seed`

### ISSUE THREAD

**Issue by Delogon** [bambinos/bambi#953]
Hi 
after spending some hours to create a reproducible analysis, I stumbled on something I think is a bug. 

When calling `model.predict()` with `sample_new_groups=True` and a `random_seed` parameter, predictions are non-deterministic. Multiple calls with the same seed produce different results, making it impossible to reproduce predictions for new groups.

## Steps to Reproduce

```python
import numpy as np
import pandas as pd
import bambi as bmb

# Create dataset with grouped data
np.random.seed(123)
data = pd.DataFrame({
    'y': np.random.randn(100),
    'x': np.random.randn(100),
    'group': np.repeat(['A', 'B', 'C', 'D'], 25)
})

# Fit model with random effects
model = bmb.Model('y ~ x + (1|group)', data)
idata = model.fit(tune=500, draws=500, random_seed=123)

# Create new data with unseen group
new_data = pd.DataFrame({
    'x': [0.5, 1.0, 1.5],
    'group': ['E', 'E', 'E']  # New group not in training data
})

# Make predictions with same random_seed multiple times
pred1 = model.predict(idata, data=new_data, kind='response_params',
                      sample_new_groups=True, random_seed=123)
pred2 = model.predict(idata, data=new_data, kind='response_params',
                      sample_new_groups=True, random_seed=123)

# Check if results are identical
mu1 = pred1.posterior['mu'].values
mu2 = pred2.posterior['mu'].values
print(f"Results identical: {np.allclose(mu1, mu2)}")
```

I think the bug is in the `_construct_u_with_new_groups()` method in `bambi/model_components.py` at line 367:

```python
rng = np.random.default_rng()
```

This method creates a new random number generator without any seed, even though the `random_seed` parameter was provided to `model.predict()`. The `random_seed` was never propagated through the call chain to reach the method that actually performs the random sampling for new groups.

I will create a pull request that fixes this, if this is not intended.

**Comment by tomicapretto** [comment]
@Delogon: thanks for both the issue report and the PR, let's continue the chat over there :)
